### PR TITLE
implement join function for fhirpath

### PR DESCRIPTION
**Pull request by webersteffen**
-added FpJoin to collectionoperators
-added join function to fp symboltable
-added two test for FpJoin

resolves FirelyTeam/fhir-net-api#1370

### CODE REVIEW

File: src/Hl7.FhirPath/FhirPath/Functions/CollectionOperators.cs [hunk 1]
```diff
@@ -87,5 +87,22 @@ public static IEnumerable<ITypedElement> Navigate(this ITypedElement element, st
...
+        public static string FpJoin(this IEnumerable<ITypedElement> collection, string separator)
+        {
+            //if the collection is empty return the empty result
+            if (!collection.Any())
+                return string.Empty;
+
+            //only join collections with string values inside
+            foreach (var typedElement in collection)
+            {
+                if (typedElement.InstanceType != "string")
```
Comment (webersteffen) [inline src/Hl7.FhirPath/FhirPath/Functions/CollectionOperators.cs:100]: @marcovisserFurore : Hi marco. At this place I didnt found any good solution to check this type better than that. In Primitive.cs the types get set hard coded as string. Maybe there is some enum or somethig that I didn´t see?

File: src/Hl7.FhirPath/FhirPath/Functions/CollectionOperators.cs [hunk 2]
```diff
@@ -112,5 +112,22 @@ public static IEnumerable<ITypedElement> Navigate(this ITypedElement element, st
...
```
Comment (marcovisserFurore) [inline src/Hl7.FhirPath/FhirPath/Functions/CollectionOperators.cs:125]: I would replace the foreach loop with this:

```
 if (!collection.All(c => c.Value is string))   
   throw Error.InvalidOperation("Join function can only be performed on string collections.");
```

File: src/Hl7.FhirPath.Tests/Tests/BasicFunctionTests.cs [hunk 3]
```diff
@@ -74,6 +76,32 @@ public void TestStringSplit()
...
+        [TestMethod]
+        public void TestStringJoin()
+        {
+            var dummy = ElementNode.CreateList("This ", "is ", "one ", "sentence", ".");
+            var result = dummy.FpJoin(string.Empty);
+            Assert.IsNotNull(result);
+            Assert.AreEqual("This is one sentence.", result);
+
+            dummy = ElementNode.CreateList();
+            result = dummy.FpJoin("");
```
[the remark this answers is not in the document]
Comment (webersteffen) [reply src/Hl7.FhirPath.Tests/Tests/BasicFunctionTests.cs]: changed it

File: src/Hl7.FhirPath/FhirPath/Functions/CollectionOperators.cs (hunk 2 above)
Comment (webersteffen) [reply src/Hl7.FhirPath/FhirPath/Functions/CollectionOperators.cs]: Thanks for the suggestion. I talked to ewout and replaced the impleentation with your suggestion.

**Pull request merged**